File: src/CoreERP.Domain/Entities/Anagrafica/Indirizzo.cs
using CoreERP.Domain.Common;

namespace CoreERP.Domain.Entities.Anagrafica;

public class Indirizzo : AuditableEntity
{
    public int AnagraficaId { get; set; }

    // Uso indirizzo (booleani, possono coesistere)
    public bool IsFatturazione { get; set; }
    public bool IsImpianto { get; set; }

    // Solo per Impianto: FTTH, FTTC, FWA
    public string? SottoTipo { get; set; }

    // Solo per Impianto: FIBERCOP, OPENFIBER, SPADHAUSEN, EOLO, etc.
    public string? Rete { get; set; }

    // Indirizzo EGON
    public string Strada { get; set; } = string.Empty;
    public string Numero { get; set; } = string.Empty;
    public string? Frazione { get; set; }
    public string Citta { get; set; } = string.Empty;
    public string Provincia { get; set; } = string.Empty;
    public string? Regione { get; set; }
    public string? CAP { get; set; }

    // Coordinate
    public double? Latitudine { get; set; }
    public double? Longitudine { get; set; }

    // Codici EGON
    public string? EgonCivico { get; set; }
    public string? EgonStrada { get; set; }
    public string? EgonLocalita { get; set; }

    // Navigation
    public Anagrafica Anagrafica { get; set; } = null!;
}

File: frontend/src/types/indirizzo.ts
export interface IndirizzoApi {
  id: number
  anagraficaId: number
  isFatturazione: boolean
  isImpianto: boolean
  sottoTipo?: string | null
  rete?: string | null
  strada: string
  numero: string
  frazione?: string | null
  citta: string
  provincia: string
  regione?: string | null
  cap?: string | null
  latitudine?: number | null
  longitudine?: number | null
  egonCivico?: string | null
  egonStrada?: string | null
  egonLocalita?: string | null
  anagraficaDenominazione?: string | null
  dataCreazione: string
}

export interface IndirizzoListItemApi {
  id: number
  anagraficaId: number
  anagraficaDenominazione: string
  isFatturazione: boolean
  isImpianto: boolean
  sottoTipo?: string | null
  rete?: string | null
  indirizzoCompleto: string
  citta: string
  provincia: string
}

export interface CreateIndirizzoRequest {
  anagraficaId: number
  isFatturazione: boolean
  isImpianto: boolean
  sottoTipo?: string | null
  rete?: string | null
  strada: string
  numero: string
  frazione?: string | null
  citta: string
  provincia: string
  regione?: string | null
  cap?: string | null
  latitudine?: number | null
  longitudine?: number | null
  egonCivico?: string | null
  egonStrada?: string | null
  egonLocalita?: string | null
}

export interface UpdateIndirizzoRequest {
  isFatturazione: boolean
  isImpianto: boolean
  sottoTipo?: string | null
  rete?: string | null
  strada: string
  numero: string
  frazione?: string | null
  citta: string
  provincia: string
  regione?: string | null
  cap?: string | null
  latitudine?: number | null
  longitudine?: number | null
  egonCivico?: string | null
  egonStrada?: string | null
  egonLocalita?: string | null
}

// EGON autocomplete types
export interface EgonComuneApi {
  egonComune: string
  comune: string
}

export interface EgonStradaApi {
  egonStrada: string
  strada: string
  comune: string
  provincia: string
  cap?: string | null
  frazione?: string | null
}

export interface EgonCivicoApi {
  egonCivico: string
  civico: string
}

export interface TipoTecnologiaApi {
  id: number
  nome: string
  descrizione?: string | null
  attivo: boolean
  ordine: number
}

export interface ReteRiferimentoApi {
  id: number
  nome: string
  descrizione?: string | null
  attivo: boolean
  ordine: number
}
